software/webapp/src/test/java/org/apache/brooklyn/entity/webapp/WebAppMigrationLiveTest.java

```diff
@@ -0,0 +1,136 @@
+
+        // Also removes scriptHeader (e.g. if doing `. ~/.bashrc` and `. ~/.profile`, then that can cause "stdin: is not a tty")
+        brooklynProperties.remove("brooklyn.ssh.config.scriptHeader");
+        mgmt = new LocalManagementContextForTests(brooklynProperties);
+        amazonProvisioningLocation = (ProvisioningLocation) mgmt.getLocationRegistry().resolve(LOCATION_SPEC);
+        super.setUp();
+    }
+
+    @DataProvider(name = "webAppEntities")
+    public Object[][] webAppEntities() {
+        Object[][] dataProviderResult;
+
+        Reflections reflections = new Reflections("org.apache.brooklyn.entity.webapp");
+        Set<Class<? extends JavaWebAppSoftwareProcess>> reflectionsSubTypesOfJavaWebAppSoftwareProcess =
+                reflections.getSubTypesOf(org.apache.brooklyn.entity.webapp.JavaWebAppSoftwareProcess.class);
+
+        List<Object[]> webAppEntities = getEntitiesFromClasses(reflectionsSubTypesOfJavaWebAppSoftwareProcess);
+        dataProviderResult = new Object[webAppEntities.size()][];
+        webAppEntities.toArray(dataProviderResult);
+
+        return dataProviderResult;
+    }
+
+
+    private List<Object[]> getEntitiesFromClasses(Set<Class<? extends JavaWebAppSoftwareProcess>> clazzSet) {
+        ArrayList<Object[]> entitiyList = new ArrayList<>();
+
+        for (Class clazz : clazzSet) {
+            if (clazz.isInterface() && !clazz.getName().contains("JavaWebAppSoftwareProcess")) {
+                EntitySpec entity = EntitySpec.create(clazz);
+                entitiyList.add(new Object[]{entity});
+            }
+
+        }
+
+
+        return entitiyList;
+    }
+
+    private String getTestWar() {
+        TestResourceUnavailableException.throwIfResourceUnavailable(getClass(), "/hello-world.war");
+        return "classpath://hello-world.war";
+    }
+
+    @Test(groups = "Live", dataProvider = "webAppEntities")
+    public void testMigration(EntitySpec<? extends JavaWebAppSoftwareProcess> webServerSpec) throws Exception {
+        final JavaWebAppSoftwareProcess server = app.createAndManageChild(webServerSpec
+                .addInitializer(new JavaWebAppMigrateEffector())
+                .configure(JavaWebAppSoftwareProcess.OPEN_IPTABLES, true)
+                .configure("war", getTestWar()));
+
+        // Run the webserver in the premigration location
+        app.start(ImmutableList.of(amazonProvisioningLocation));
+        EntityTestUtils.assertAttributeEqualsEventually(server, Attributes.SERVICE_UP, Boolean.TRUE);
+
+        // Check if the WebServer is reachable in the premigration location
+        String preMigrationUrl = server.getAttribute(JavaWebAppSoftwareProcess.ROOT_URL);
+        HttpTestUtils.assertUrlReachable(preMigrationUrl);
+
+        // Start the migration process
+        Effectors.invocation(AllowsMigration.MIGRATE, MutableMap.of("locationSpec", LOCATION_SPEC), server).asTask().blockUntilEnded();
```

Comment:
Give a `Duration` timeout argument to `blockUntilEnded` just in case something goes wrong.